opencti-platform/opencti-front/src/private/components/common/lists/FilterAutocomplete.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { Dispatch, FunctionComponent, SyntheticEvent, useState } from 'react';
+import TextField from '@mui/material/TextField';
+import * as R from 'ramda';
+import Autocomplete from '@mui/material/Autocomplete';
+import makeStyles from '@mui/styles/makeStyles';
+import { AutocompleteValue } from '@mui/material';
+import ItemIcon from '../../../../components/ItemIcon';
+import { useFormatter } from '../../../../components/i18n';
+import useSearchEntities from '../../../../utils/filters/useSearchEntities';
+import { Theme } from '../../../../components/Theme';
+import SearchScopeElement from './SearchScopeElement';
+import { onlyGroupOrganization } from '../../../../utils/filters/filtersUtils';
+import { HandleAddFilter } from '../../../../utils/hooks/useLocalStorage';
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+    color: theme.palette.primary.main,
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+}));
+
+interface OptionValue {
+  value: string
+  label: string
+  type: string
+  group?: string
+  color?: string
+}
+
+interface FilterAutocompleteProps {
+  filterKey: string
+  defaultHandleAddFilter: HandleAddFilter
+  inputValues: Record<string, string | Date>
+  setInputValues: Dispatch<Record<string, string | Date>>
+  availableEntityTypes: Array<string>
```

Comment:
Prefer string[] instead of Array<string> for clarity

Reply:
Can we add it to the filigran code style ? Or it's a Typescript convention ?

Reply:
Fix